tests/e2e/vsphere_volume_expansion.go

```diff
@@ -199,8 +238,440 @@ var _ = ginkgo.Describe("Volume Expansion Test", func() {
+		err = fmt.Errorf("Received wrong disk size after volume expansion. Expected: %d Actual: %d", newSizeInMb, queryResult.Volumes[0].BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails).CapacityInMb)
+	}
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	framework.ExpectNoError(waitForPVCToCompleteResizing(client, namespace, pvclaim.Name, poll, totalResizeWaitPeriod))
+	ExpectedSizeInMb := int64(12288)
+	framework.ExpectNoError(verifyPVCSizeAfterResize(f, client, namespace, pvclaim.Name, ExpectedSizeInMb))
+
+	ginkgo.By("Verify filesystem size for mount point /mnt/volume1")
+	fsSize, err := getFSSizeMb(f, pod)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	// Filesystem size may be smaller than the size of the block volume
+	// so here we are checking if the new filesystem size is greater than
+	// the original volume size as the filesystem is formatted.
+
+	gomega.Expect(fsSize).Should(gomega.BeNumerically(">", diskSizeInMb), fmt.Sprintf("error updating filesystem size for %q. Resulting filesystem size is %d", pvclaim.Name, fsSize))
+	ginkgo.By(fmt.Sprintf("File system resize finished successfully to %d", fsSize))
+}
+
+//verifyPVCSizeAfterResize this method verifies PVC size with the expected size
+func verifyPVCSizeAfterResize(f *framework.Framework, client clientset.Interface, namespace string, pvcName string, expectedsize int64) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	pvclaim, err := client.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, pvcName, metav1.GetOptions{})
+	ginkgo.By("Verifying PVC size requested in volume expansion is honored")
+	currentPvcSize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+	size := currentPvcSize.DeepCopy()
+	framework.Logf("Actual size : %d", size)
+	actualSize := convertBytesToMb(size)
+	framework.Logf("Actual size after converting to MB : %d", actualSize)
+	if actualSize != expectedsize {
+		err = fmt.Errorf("Received wrong disk size after volume expansion. Expected: %d Actual: %d", expectedsize, actualSize)
+		return err
+	}
+	return nil
+}
+
+//convertBytesToMb this methos converts Bytes to MB
+func convertBytesToMb(size resource.Quantity) int64 {
+	actualSize, _ := size.AsInt64()
+	actualSize = actualSize / (1024 * 1024)
+	return actualSize
+}
+
+//createStaticPVC this method creates static PVC
+func createStaticPVC(f *framework.Framework, client clientset.Interface, namespace string, defaultDatastore *object.Datastore, pandoraSyncWaitTime int) (string, *v1.PersistentVolumeClaim, *v1.PersistentVolume, *storagev1.StorageClass) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	curtime := time.Now().Unix()
+	randomValue := rand.Int()
+	curtimeinstring := strconv.FormatInt(curtime, 10)
+	val := strconv.FormatInt(int64(randomValue), 10)
+	val = string(val[1:3])
+	curtimestring := strconv.FormatInt(curtime, 10)
+	pvcName := "pvc-" + curtimestring + val
```

Comment:
we don't have to do this, you can use a simple string like `"static-pvc-"`, since pvc name that we pass to `getPVCSpecWithPVandStorageClass()` is assigned to `GenerateName`, k8s will add a unique string at the end of the pvc name.